client/browser/src/browser-extension/scripts/backgroundPage.main.ts

```diff
@@ -53,33 +53,53 @@ initSentry('background')
 const IsProductionVersion = !getExtensionVersion().startsWith('0.0.0')
 
 /**
- * For each tab, we store a flag if we know that we are on a private
- * repository that has not been added to Cloud (+ the extension
- * points to Cloud). The content script notifies the background page if it has
- * experienced a private code on Cloud error by `notifyPrivateCloudError` message.
+ * For each tab, we store a flag if we know that we are on:
+ * - a private repo not synced with Sourcegraph Cloud when the latter is the active Sourcegrapg URL
+ * - a repo not added to the other than Cloud Sourcegraph instance (+ the extension * points to this instance).
+ * The content script notifies the background page if it has experienced this kind of an error
+ * by sending `notifyRepoSyncError` message.
  */
-const tabPrivateCloudErrorCache = (() => {
-    const cache = new Map<number, boolean>()
-    const subject = new Subject<ReadonlyMap<number, boolean>>()
+const tabRepoSyncErrorCache = (() => {
+    const cache = new Map<number, Map<string, boolean>>()
+    const subject = new Subject<ReadonlyMap<number, Map<string, boolean>>>()
     return {
         observable: subject.asObservable(),
         /**
-         * Update the background page's cache of which tabs have experienced a
-         * private code on Cloud error.
+         * Update the background page's cache of which tabs have experienced either a
+         * private code on Cloud or not synced repo on other than Cloud Sourcegrpah instance error.
          */
-        setTabHasPrivateCloudError(tabId: number, hasPrivateCloudError: boolean): void {
-            if (!hasPrivateCloudError) {
-                // An absent value is equivalent to being false; so we can delete it.
+        setTabHasRepoSyncError(tabId: number, sourcegraphURL: string, hasRepoSyncError: boolean): void {
+            if (sourcegraphURL) {
+                const record = cache.get(tabId)
+
+                if (record) {
+                    if (hasRepoSyncError) {
+                        record.set(sourcegraphURL, true)
+                    } else {
+                        record.delete(sourcegraphURL)
+                    }
+
+                    if (record.size === 0) {
+                        cache.delete(tabId)
+                    }
+                } else if (hasRepoSyncError) {
+                    cache.set(tabId, new Map().set(sourcegraphURL, true))
+                }
+            } else {
                 cache.delete(tabId)
             }
```

Comment:
Just minor rearranging to decrease if/else nesting:
```suggestion
            if (sourcegraphURL) {
                let record = cache.get(tabId)

                if (!record) {
                    record = new Map()
                    cache.set(tabId, record)
                }

                if (hasRepoSyncError) {
                    record.set(sourcegraphURL, true)
                } else if (record.size === 0) {
                    cache.delete(tabId)
                } else {
                    record.delete(sourcegraphURL)
                }
            } else {
                cache.delete(tabId)
            }
```

Reply:
Nice refactoring!